engine/src/parse_file.rs

```diff
@@ -292,13 +386,30 @@ impl ParsedFile {
 impl ToTokens for ParsedFile {
     fn to_tokens(&self, tokens: &mut TokenStream) {
         for seg in &self.0 {
-            match seg {
-                Segment::Other(item) => item.to_tokens(tokens),
-                Segment::Autocxx(autocxx) => {
-                    let these_tokens = autocxx.generate_rs();
-                    tokens.extend(these_tokens);
+            seg.to_tokens(tokens)
+        }
+    }
+}
+
+impl ToTokens for Segment {
+    fn to_tokens(&self, tokens: &mut TokenStream) {
+        match self {
+            Segment::Other(item) => item.to_tokens(tokens),
+            Segment::Autocxx(autocxx) => {
+                let these_tokens = autocxx.generate_rs();
+                tokens.extend(these_tokens);
+            }
+            Segment::Cxx(cxxbridge) => cxxbridge.to_tokens(tokens),
+            Segment::Mod(segments, (brace, itemmod)) => {
+                let mut mod_items = Vec::with_capacity(segments.len());
```

Comment:
I have a slight preference for some kind of `let mod_items = segments.iter().map(...)` thing but no biggie. Only if you feel like adjusting it.

Reply:
Yep, that's nicer. That's my C++ showing...